src/controllers/profile.js

```diff
@@ -1,6 +1,60 @@
-const Product = require('../models/product');
+const bcrypt = require('bcrypt');
+const { validationResult } = require('express-validator');
 const ErrorResponse = require('../utils/errorResponse');
+const User = require('../models/user');
+
+const Product = require('../models/product');
+
+exports.getMyProfile = async (req, res) => {
+  res.json({ success: true, data: req.user });
+
+  // This is an example of a successful response
+};
+exports.updateProfile = async (req, res, next) => {
+  const validationErrors = validationResult(req);
+  if (!validationErrors.isEmpty()) {
+    return res
+      .status(400)
+      .json({ success: false, errors: validationErrors.array() });
+  }
 
+  let user = await User.findById(req.user._id);
+
+  const userexist = await User.findOne({ email: req.body.email });
+  if (userexist) {
+    return next(new ErrorResponse('Email already exists!', 400));
+  }
```

Comment:
Since email field is optional, `req.body.email`, could be null. So you need to check that first, then check the existence of any users.

Reply:
if ('email' in req.body) {
  ...
}